src/backend/base/langflow/components/tools/beautiful_soup_scrapper.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from urllib.parse import urljoin, urlparse
+
+import requests
+from bs4 import BeautifulSoup
+from langchain.tools import StructuredTool
+from pydantic import BaseModel, Field
+
+from langflow.base.langchain_utilities.model import LCToolComponent
+from langflow.field_typing import Tool
+from langflow.inputs import DictInput, IntInput, MessageTextInput
+from langflow.schema import Data
+
+logging.basicConfig(level=logging.ERROR)
+
+HTTP_SUCCESS_CODE = 200
+
+
+class BeautifulSoupScrapeApiComponent(LCToolComponent):
+    display_name = "Beautiful Soup Scraper"
+    description = "Scrape web content using BeautifulSoup."
+    name = "beautiful_soup_scraper"
+    documentation = "https://www.crummy.com/software/BeautifulSoup/bs4/doc/"
+    icon = "BeautifulSoup"
+
+    inputs = [
+        MessageTextInput(
+            name="url",
+            display_name="URL",
+            info="The URL to scrape.",
+        ),
+        IntInput(
+            name="timeout",
+            display_name="Timeout",
+            info="Timeout in seconds for the request.",
+            value=300,
+        ),
+        DictInput(
+            name="page_options",
+            display_name="Page Options",
+            info="The page options to send with the request.",
+            advanced=True,
+        ),
+        DictInput(
+            name="extractor_options",
+            display_name="Extractor Options",
+            info="The extractor options to send with the request.",
+            advanced=True,
+        ),
+    ]
+
+    class BeautifulSoupScrapeApiSchema(BaseModel):
+        url: str = Field(..., description="The URL to scrape")
+        timeout: int = Field(default=10, description="Timeout in seconds for the request")
+        page_options: dict | None = Field(default=None, description="The page options to send with the request")
+        extractor_options: dict | None = Field(
+            default=None, description="The extractor options to send with the request"
+        )
```

Comment:
Hey @joaoguilhermeS 

Now we can define classes outside the Component.

Reply:
@ogabrielluiz I did try this and it was causing error.

How about we make this into a data component with  tool mode available?
or even try using URL compoment as tool.